server/src/main/java/org/apache/iotdb/db/query/executor/LastQueryExecutor.java

```diff
@@ -299,7 +317,42 @@ public void write(TimeValuePair pair) {
     }
   }
 
+  private static class IDTableLastCacheAccessor implements LastCacheAccessor {
+
+    private PartialPath fullPath;
+
+    IDTableLastCacheAccessor(PartialPath seriesPath) {
+      fullPath = seriesPath;
+    }
+
+    @Override
+    public TimeValuePair read() {
+      try {
+        IDTable table = IDTableManager.getInstance().getIDTable(fullPath.getDevicePath());
```

Comment:
this is a global lock, could we use the StorageGroupProcessor to get the IDTable?

Reply:
Sure, fixed~